docs/forms-in-wagtail.md

```diff
@@ -0,0 +1,44 @@
+# Including forms into Wagtail page context
```

Comment:
> more internal details of our development process and less appropriate for developers trying to use the code

I think I understand where you're coming from. However, is our documentation supposed to be restrictively about code use? The first two paragraphs serve as an explanation for its use and purpose, not to criticize Wagtail. 

I find it wholly better to introduce something (especially something that changes a lot) with more explanation than just "here's how you use it", because it lets others in on the problem it is attempting to solve. That way down the road if the problem goes away or a better solution can be had, it would be easier to refactor since anyone can track down how and why it's there in the first place.

I'm completely open to scrapping this in favor of brevity, I just think giving something context is a generally good thing.